allennlp/common/logging.py

```diff
@@ -0,0 +1,44 @@
+"""
+A custom 'logging.Logger' that has additional {warning,debug,etc.}_once() methods
+which allow for logged methods to be logged only once.
+"""
+import logging
+
+
+class AllenNlpLogger(logging.Logger):
+    """
+    A custom subclass of 'logging.Logger' that keeps a set of messages to
+    implement {debug,info,etc.}_once() methods.
+    """
+
+    def __init__(self, name):
+        super().__init__(name)
+        self.msgs = set()
```

Comment:
Can you call this variable `self._seen_msgs`, or something with an underscore? We generally use underscores to mark "private" variables.